### PR TITLE
wip

### PR DESCRIPTION
**Pull request by adelura**
Hi, just opened pull request to know your opinion about that change. It's a breaking change for sure. My motivation was to exclude mixins and functions from the output. Also, I wanted to expose more information which is a path to file where a variable was defined. It's because I wanted to exclude "vendor" unused variables from the output.

Just curious about your opinion.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,6 +19,8 @@
   },
   "devDependencies": {
     "chai": "^4.1.2",
+    "lodash.flatten": "^4.4.0",
+    "lodash.values": "^4.3.0",
```
Comment (adelura) [inline package.json]: This shouldn't be here.

**Comment by robertknight** [comment]
Hello @adelura, thank-you for your effort.  It will be much easier for me to give feedback and +1  or -1 if you could make the PR description an explanation of:

- What problem / need to you are addressing
- What the change is conceptually (from the point of view of a user of the tool)
- If the code change is non-trivial, a high level explanation of the implementation is also useful

**Comment by MatteoWebDeveloper** [comment]
Hi Robert, 
I work with Artur, I will explain what we tried to achieve.

Our codebase include some modules which store reusable scss code like: 
common variables, mixins, components, ect..

The reason we had to modularize our scss is because we have multiple apps and we want to reuse our common style.

app.scss
```scss
@import "./settings/settings";

@import "node_modules/styles-base/settings";

@import "node_modules/styles-components/components";
@import "./components/app-specific-style";
```

the problem we had with sass-unused was we could not exclude vendor variables/mixins/functions, for instance:

app.scss
```scss
/* $ven-s-color is not used */
```

node_modules/styles-base/settings
```scss
$ven-s-color: blue !default;
```

will list it as a unsed variables.

The only way to exclude that variable was to first make sass-unsed to output a more informative data:
https://github.com/robertknight/sass-unused/pull/2/files#diff-d55c7ae6346eade945088fe4d357da5bR133

output:
```js
{ 
    vars: [
         { path: '.../app/node_modules/styles-base/settings/settings.scss', name: 'ven-s-global-color-1' },  
         { path: '.../app/node_modules/styles-base/settings/settings.scss', name: 'ven-s-global-spacing-1' } ,
         { path: '.../app/styles/settings/settings.scss', name: 'app-s-app-specific-varaible' }   
     ], 
    mixins: [], 
    functions: [] }
```

Now we know for each variables where it comes from and we can create another tool which will filter all the vendor unused variables.

We built another tool which use sass-unused and we plan to make it open source

File: lib/find-unused.js [hunk 2]
```diff
@@ -130,31 +130,32 @@ function findUnused(srcFiles, resolver) {
...
-  return [
-    ...unusedVars,
-    ...unusedMixins,
-    ...unusedFunctions,
-  ];
+  return {
+    vars: unusedVars,
+    mixins: unusedMixins,
+    functions: unusedFunctions
+  };
```
Comment (MatteoWebDeveloper) [inline lib/find-unused.js:158]: We changed to object so we can differentiate between var, mixin and function.

File: lib/find-unused.js [hunk 3]
```diff
@@ -130,31 +130,32 @@ function findUnused(srcFiles, resolver) {
   let usedFunctions = new Set();
 
   srcFiles.forEach(path => {
+    const objectWithPath = name => { return { name, path }; };
```
Comment (MatteoWebDeveloper) [inline lib/find-unused.js:133]: now each item can now output 

```js
{ path: '.../app/node_modules/styles-base/settings/settings.scss', name: 'ven-s-global-color-1' }
```

**Pull request closed without merging**